Doc/tutorial/classes.rst

```diff
@@ -337,12 +337,11 @@ code will print the value ``16``, without leaving a trace::
    print(x.counter)
    del x.counter
 
-The other kind of instance attribute reference is a *method*. A method is a
-function that "belongs to" an object.  (In Python, the term method is not unique
-to class instances: other object types can have methods as well.  For example,
-list objects have methods called append, insert, remove, sort, and so on.
-However, in the following discussion, we'll use the term method exclusively to
-mean methods of class instance objects, unless explicitly stated otherwise.)
+The other kind of instance attribute reference is a *method*. In the following
+discussion, we’ll use the term method exclusively to mean methods of class
+instance objects, unless explicitly stated otherwise. (In other words, we are
+not referring to built-in methods such as ``alist.append`` where ``alist`` is
+a :class:`list` object.)
```

Comment:
This paragraph is _very_ old. It appears to have remained in this tutorial largely unchanged since it was originally added in 1992, in https://github.com/python/cpython/commit/5e0759d351ed6ceb3d0c28475df214d5e5ffa626.

I don't think that the original purpose of this paragraph was to point out that the type of `alist.append` is a different class to the type of a user-defined method on a user-defined class. In very early versions of Python, many builtin types such as `list` were not classes at all -- they were more primitive types that operated according to a different object model. (See https://stackoverflow.com/a/54873/13990016 for more details. This changed in Python 2.2, originally released in 2001.) I think this paragraph was originally intended to clarify that even objects that were not instances of classes, such as lists, could also have methods.

Nowadays, all objects are instances of classes, and it is not possible to create types that do not inherit from `object`. It's been this way for a long time now. As such, I think this paragraph is now largely redundant. Rather than making the edit you propose here, I would simply delete most of this paragraph altogether.

```suggestion
The other kind of instance attribute reference is a *method*. A method is a
function that "belongs to" an object.
```

Reply:
I thought about suggesting this originally, but https://docs.python.org/3/reference/datamodel.html#built-in-methods makes it sound like there's still some difference between built-in methods and methods on user defined classes (besides built-ins being defined in C that is). As mentioned in https://github.com/python/cpython/issues/119016#issuecomment-2113375783, the types are different at least.

Does the last paragraphs in particular of both 9.3.4 (https://docs.python.org/3/tutorial/classes.html#method-objects) and 9.7 (https://docs.python.org/3/tutorial/classes.html#odds-and-ends) apply equally to built-in methods? If not, there might still be some confusion.

Reply:
Yes, builtin methods are a different type to pure-Python methods, but they're meant to be entirely duck-type compatible. Any glaring differences between them are arguably bugs. Off the top of my head, I don't think any of the paragraphs you link to there are different when it comes to builtin methods.